cobalt/browser/h5vcc_runtime/deep_link_manager.h

```diff
@@ -0,0 +1,64 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef COBALT_BROWSER_H5VCC_RUNTIME_DEEP_LINK_MANAGER_H_
+#define COBALT_BROWSER_H5VCC_RUNTIME_DEEP_LINK_MANAGER_H_
+
+#include <string>
+
+#include "base/no_destructor.h"
+#include "base/threading/thread_checker.h"
+
+namespace cobalt {
+namespace browser {
+
+//  This class is a singleton that provides a central point for storing and
+//  retrieving the deep link URL that was used to launch the application.  It
+//  handles both initial deep links (available at app startup) and warm start
+//  deep links (received while the app is already running). It ensures that
+//  the deep link is delivered exactly once to the JavaScript side. This
```

Comment:
nit: "any deep link is delivered exactly once" or "deep links are delivered exactly once"